File: 0_1.baekjoon/brute-force/python/BOJ1107.py
from sys import stdin
from collections import defaultdict

readline = stdin.readline

INF = 1000000
N = int(readline())
M = int(readline())
out_of_control = set()

for num in list(map(int, readline().split())):
    out_of_control.add(num)

def solve():
    result = abs(N - 100)

    min_diff_num = get_min_diff_num()

    result = min(result, abs(min_diff_num - N) + len(str(min_diff_num)))
    print(result)

def get_min_diff_num():
    min_diff = INF
    num = INF
    for i in range(INF):
        diff = abs(i - N)
        if check(i) and diff < min_diff:
            min_diff = diff
            num = i

    return num

def check(num):
    if num == 0:
        return num not in out_of_control

    temp = num

    while temp != 0:
        remain = temp % 10

        if remain in out_of_control:
            return False

        temp //= 10

    return True

solve()

File: 0_1.baekjoon/brute-force/java/src/BOJ1107.java
import java.util.*;
import java.io.*;

public class BOJ1107 {
    static final int INF = 1000000;
    static int N, M;
    static boolean[] outOfControl;
    static int result = INF;
    public static void main(String[] args) throws NumberFormatException, IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());
        outOfControl = new boolean[10];

        if (M != 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
    
            for (int i=0;i<M;i++) {
                outOfControl[Integer.parseInt(st.nextToken())] = true;
            }
        }

        result = Math.abs(100 - N);
        int minDiffNum = getMinDiffNum();
        result = Math.min(result, Math.abs(minDiffNum - N) + getDigits(minDiffNum));
        System.out.println(result);
    }
    private static int getDigits(int num) {
        int digits = 1;
        while ((num / 10) != 0) {
            num /= 10;
            digits++;
        }

        return digits;
    }

    private static int getMinDiffNum() {
        int minDiff = INF;
        int returnValue = INF; 

        for (int i=0;i<INF;i++) {
            int diff = Math.abs(N - i);

            if (check(i) && diff < minDiff) {
                minDiff = diff;
                returnValue = i;
            }
        }
        return returnValue;
    }

    private static boolean check(int num) {
        if (num == 0) {
            return !outOfControl[0];
        }

        int temp = num;

        while (temp != 0) {
            int remain = temp % 10;

            if (outOfControl[remain])
                return false;

            temp /= 10;
        }
        return true;
    }

}
